### PR TITLE
Pk3 file support

### PR DESCRIPTION
**Pull request by 30-something-programmer**
Almost there, still need to add in auto-checking for versioning and to put the pokemon dex number at the beginning of the file.

### CODE REVIEW

**Comment by pekempy** [comment]
Tip: For the Pokedex number you can likely leverage modules/data/species.json to find the natdex number of the 'mon and then left pad with zeroes

File: modules/pokemon.py [hunk 1]
```diff
@@ -758,6 +761,7 @@ class Pokemon:
 
     def __init__(self, data: bytes):
         self.data = data
+        self.save()
```
[the remark this answers is not in the document]
Comment (30-something-programmer) [reply modules/pokemon.py:764]: gotcha! Working on this now

File: modules/pokemon.py [hunk 2]
```diff
@@ -1140,6 +1144,61 @@ def wurmple_evolution(self) -> Literal["silcoon", "cascoon"]:
...
+
+        # Pokemon is shiny and user wants to save shiny pokemon
+        elif config["general"]["save_shiny_pokemon"] and self.is_shiny:
+            prep_pk_for_save(self)
+
+        # Pokemon suits custom paramaters and user wants to save it
+        else:
+            from profiles.customcatchfilters import custom_catch_filters
+            if config["general"]["save_custom_pokemon"] and custom_catch_filters(self):
+                prep_pk_for_save(self)
```
[the remark this answers is not in the document]
Comment (30-something-programmer) [reply modules/pokemon.py:1198]: Happy to follow! I've added the function into stats and removed it from a method of Pokemon.

**Comment by 30-something-programmer** [comment]
Commit created - Pk3 files generated.
This should resolve the issues laid out. On the issue of PC/party backup. The SAV file within /profiles/[PROFILE] already provides this functionality?

**Comment by pekempy** [comment]
> Commit created - Pk3 files generated.
> This should resolve the issues laid out. On the issue of PC/party backup. The SAV file within /profiles/[PROFILE] already provides this functionality? 

I believe the concept for the backup was a way to bulk export the .pk3 files, not sure if pkhex provides that functionality - if it does, I suppose it could be left out of the PR

File: profiles/general.yml [hunk 3]
```diff
@@ -8,3 +8,10 @@ bot_mode: manual
 # Starter - used when `bot_mode` is set to `starters`
 # `Treecko`, `Torchic`, `Mudkip`, `Bulbasaur`, `Charmander`, `Squirtle`, `Chikorita`, `Cyndaquil`, `Totodile`
 starter: Mudkip
+
+
+# Select what pokemon to save as a pk file
+# save_all_pokemon overrides the other two
+save_all_pokemon: true
```
Comment (pekempy) [inline profiles/general.yml:15]: nit: I would probably have `save_all_pokemon` as `false` by default, I can't imagine people want 3 million .pk3 files, primary use case would be for shinies/custom catch

Comment (30-something-programmer) [reply profiles/general.yml]: I'll change it :)

**Comment by 40Cakes** [comment]
Nice job on this, works like a charm! 👌

**Pull request merged**